pkg/controller/pcideviceclaim/pcideviceclaim_controller.go

```diff
@@ -335,45 +290,139 @@ func (h Handler) reconcilePCIDeviceClaims(name string, pdc *v1beta1.PCIDeviceCla
 		return pdc, nil
 	}
 
-	if pdc.Spec.NodeName == h.nodeName && !pdc.Status.PassthroughEnabled {
-		newPdc, err := h.attemptToEnablePassthrough(pdc)
-		return newPdc, err
+	if pdc.Spec.NodeName != h.nodeName || pdc.Status.PassthroughEnabled {
+		return pdc, nil
+	}
+
+	// Get the PCIDevice object for the PCIDeviceClaim
+	pd, err := h.getPCIDeviceForClaim(pdc)
+	if pd == nil {
+		return pdc, err
+	}
+
+	// Enable PCI Passthrough on the device by binding it to vfio-pci driver
+	newPdc, err := h.attemptToEnablePassthrough(pd, pdc)
+
+	// Find the DevicePlugin
+	resourceName := pd.Status.ResourceName
+	dp := deviceplugins.Find(
+		resourceName,
+		h.devicePlugins,
+	)
+
+	h.permitHostDeviceInKubeVirt(pd)
+
+	// If the DevicePlugin can't be found, then create it
+	if dp == nil {
+		logrus.Infof("Creating DevicePlugin: %s", resourceName)
+		pds := []*v1beta1.PCIDevice{pd}
+		dp = deviceplugins.Create(resourceName, pdc, pds)
+		h.devicePlugins[resourceName] = dp
+		// Start the DevicePlugin
+		if newPdc.Status.PassthroughEnabled && !dp.Started() {
+			h.startDevicePlugin(pd, newPdc, dp)
+		}
+	} else {
+		// Add the Device to the DevicePlugin
+		dp.AddDevice(pd, pdc)
 	}
+	return newPdc, err
+}
 
-	return pdc, nil
+func (h Handler) permitHostDeviceInKubeVirt(pd *v1beta1.PCIDevice) error {
+	logrus.Infof("Adding %s to KubeVirt list of permitted devices", pd.Name)
+	kv, err := h.virtClient.KubeVirt(DefaultNS).Get(KubevirtCR, &v1.GetOptions{})
+	if err != nil {
+		msg := fmt.Sprintf("cannot obtain KubeVirt CR: %v", err)
+		return errors.New(msg)
+	}
+	kvCopy := kv.DeepCopy()
+	if kv.Spec.Configuration.PermittedHostDevices == nil {
+		kvCopy.Spec.Configuration.PermittedHostDevices = &kubevirtv1.PermittedHostDevices{
+			PciHostDevices: []kubevirtv1.PciHostDevice{},
+		}
+	}
+	permittedPCIDevices := kvCopy.Spec.Configuration.PermittedHostDevices.PciHostDevices
+	resourceName := pd.Status.ResourceName
+	// check if device is currently permitted
+	var devPermitted bool = false
+	for _, permittedPCIDev := range permittedPCIDevices {
+		if permittedPCIDev.ResourceName == resourceName {
+			devPermitted = true
+			break
+		}
+	}
+	if !devPermitted {
+		vendorId := pd.Status.VendorId
+		deviceId := pd.Status.DeviceId
+		devToPermit := kubevirtv1.PciHostDevice{
+			PCIVendorSelector:        fmt.Sprintf("%s:%s", vendorId, deviceId),
+			ResourceName:             resourceName,
+			ExternalResourceProvider: false,
```

Comment:
for the plugin to be leverage the ExternalResourceProvider should be true

```suggestion
			ExternalResourceProvider: true,
```